src/util/ConfigValMap.h

```diff
@@ -162,7 +162,10 @@ class ConfigValT : public ConfigVal {
 
 protected:
     ConfigValT(std::string const& section, std::string const& name, bool required, T defVal, bool hidden)
-            : ConfigVal(section, name, required, hidden), _val(defVal) {}
+            : ConfigVal(section, name, required, hidden) {
+        _val = defVal;
+        _defVal = defVal;
+    }
```

Comment:
I ran into the same problem in the branch. I changed the code to (also making the default const)
```
protected:
    ConfigValT(std::string const& section, std::string const& name, bool required, T defVal, bool hidden)
            : ConfigVal(section, name, required, hidden), _defVal(defVal), _val(_defVal) {}

private:
    T const _defVal;  ///< Default value for the item this class is storing.
    T _val;           ///< Value for the item this class is storing.
```

Reply:
Good point! I will do the same in my PR before merging it into the main branch.